<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>30</y>
      <width>243</width>
      <height>274</height>
     </rect>
    </property>
    <property name="title">
     <string>Sinh Khóa RSA</string>
    </property>
    <widget class="QPushButton" name="btnGenerateKeys">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Sinh khóa</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>P:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Q:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>N: </string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Phi(n):</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>190</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Khóa Công khai:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>220</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Khóa Riêng:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPrivateKey">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>220</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPrimeQ">
     <property name="geometry">
      <rect>
       <x>121</x>
       <y>91</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPublicKey">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>190</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPhiN">
     <property name="geometry">
      <rect>
       <x>121</x>
       <y>151</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtN">
     <property name="geometry">
      <rect>
       <x>121</x>
       <y>121</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="txtPrimeP">
     <property name="geometry">
      <rect>
       <x>121</x>
       <y>61</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>20</y>
      <width>401</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Mã Hóa</string>
    </property>
    <widget class="QTextEdit" name="txtPlainText">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>40</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnEncrypt">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>120</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Mã Hóa</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="txtEncryptedMessage">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>160</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Bản rõ</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Bản mã</string>
     </property>
    </widget>
    <widget class="QSplitter" name="splitter">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>90</y>
       <width>179</width>
       <height>24</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <widget class="QLabel" name="label_11">
      <property name="text">
       <string>Khóa Công Khai:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtPublickey_1">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="btnSaveFile">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>170</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Xuất file</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnReadFile">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>50</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Đọc file</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>250</y>
      <width>401</width>
      <height>271</height>
     </rect>
    </property>
    <property name="title">
     <string>Giải mã</string>
    </property>
    <widget class="QTextEdit" name="txtEncryptedInput">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>40</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnDecrypt">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>140</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Giải mã</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="txtDecryptedMessage">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>170</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Bản mã</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>180</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Bản rõ</string>
     </property>
    </widget>
    <widget class="QSplitter" name="splitter_2">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>100</y>
       <width>154</width>
       <height>24</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <widget class="QLabel" name="label_12">
      <property name="text">
       <string>Khóa Riêng:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtPrivateKey_1">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="btnReadFile_2">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>50</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Đọc file</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSaveFile_2">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>180</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Xuất file</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
